### PR TITLE
Fix load collection failed after drop partition

### PR DESCRIPTION
**Pull request by bigsheeper**
issue: https://github.com/milvus-io/milvus/issues/24486

/kind bug

### CODE REVIEW

File: internal/querycoordv2/meta/collection_manager.go [hunk 1]
```diff
@@ -126,6 +127,13 @@ func (m *CollectionManager) Recover(broker Broker) error {
 	}
 
 	for _, collection := range collections {
+		// Dropped collection should be deprecated
+		_, err = broker.GetCollectionSchema(context.Background(), collection.GetCollectionID())
```
Comment (yah01) [inline internal/querycoordv2/meta/collection_manager.go:131]: this is a performance bottleneck if many collections loaded, may optimize it by concurrently recover collections

File: internal/querycoordv2/meta/collection_manager.go [hunk 2]
```diff
@@ -142,15 +150,34 @@ func (m *CollectionManager) Recover(broker Broker) error {
 	}
 
 	for collection, partitions := range partitions {
+		existPartitions, err := broker.GetPartitions(context.Background(), collection)
```
Comment (yah01) [inline internal/querycoordv2/meta/collection_manager.go:153]: Same as line 131

File: internal/querycoordv2/meta/collection_manager.go (hunk 1 above)
Comment (bigsheeper) [reply internal/querycoordv2/meta/collection_manager.go:131]: ddl requests (including DescribeCollection and ShowPartition) are executed serially in rootcoord

File: internal/querycoordv2/meta/coordinator_broker.go [hunk 3]
```diff
@@ -96,12 +103,17 @@ func (broker *CoordinatorBroker) GetPartitions(ctx context.Context, collectionID
...
-	resp, err := broker.rootCoord.ShowPartitionsInternal(ctx, req)
+	resp, err := broker.rootCoord.ShowPartitions(ctx, req)
 	if err != nil {
 		log.Warn("showPartition failed", zap.Int64("collectionID", collectionID), zap.Error(err))
 		return nil, err
 	}
 
+	statusErr := common.NewStatusError(resp.Status.ErrorCode, resp.Status.Reason)
+	if common.IsCollectionNotExistError(statusErr) {
+		return nil, merr.WrapErrCollectionNotFound(collectionID)
```
Comment (longjiquan) [inline internal/querycoordv2/meta/coordinator_broker.go:114]: Please add ut for this case.

File: internal/querycoordv2/meta/collection_manager.go [hunk 4]
```diff
@@ -142,15 +150,34 @@ func (m *CollectionManager) Recover(broker Broker) error {
...
+		if errors.Is(err, merr.ErrCollectionNotFound) {
+			log.Info("skip dropped collection during recovery", zap.Int64("collection", collection))
+			m.store.ReleaseCollection(collection)
+			continue
+		}
```
Comment (longjiquan) [inline internal/querycoordv2/meta/collection_manager.go:158]: How to handle other exceptions?

**Review by longjiquan — CHANGES_REQUESTED** [review]

**Pull request merged**